### PR TITLE
Change contribution guide and readme

### PR DESCRIPTION
**Pull request by wmontwe**
### :tophat: What is the goal?

Document the project

Bump Android Gradle Plugin 4.2.0 -> 4.2.1

### :thinking: DOD Checklist

- [ ] Code style and naming conventions met
- [ ] Test written and passing
- [ ] Continuous Integration build passing
- [ ] Cross platform testing done on Android and iOS
- [ ] Code peer-reviewed
- [ ] Documentation updated
- [ ] Changelog updated
- [ ] Acceptance criteria met

### CODE REVIEW

File: DEVELOPMENT.adoc [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+Make sure you are in the project root and install the correct Python version which is derived from `.python-version` file:
+
+[source,bash]
+----
+pyenv install
+----
+
+=== Android Studio
+
+For better interop with iOS you may install the link:https://plugins.jetbrains.com/plugin/14936-kotlin-multiplatform-mobile[KMM plugin for Android Studio]]. It let's you run, test and debug shared code on Android and iOS without switching the IDEs.
```
Comment (bitPogo) [inline DEVELOPMENT.adoc:101]: `link:https://plugins.jetbrains.com/plugin/14936-kotlin-multiplatform-mobile[KMM plugin for Android Studio]`

File: DEVELOPMENT.adoc [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+== Release
+
+A release branches of from `main` branch with following pattern:
+
+* `release/{major}.{minor}(.{patch})/prepare-{major}.{minor}.{patch}` while patch is optional
+
+Add following changes:
+* [ ] Update CHANGELOG.adoc by creating a new Unreleased section and change current unreleased to release version
+* [ ] Update README.adoc `fhir-sdk-version` to release version
```
Comment (bitPogo) [inline DEVELOPMENT.adoc:147]: Should that be a checkbox list or is this correct displayed?

File: DEVELOPMENT.adoc [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+== Develop
+
+The source is divided into supporting code for JSON parsing and establishing the FHIR primitive type system (String, Dates, Integer, etc.) for link:http://hl7.org/fhir/STU3/datatypes.html[FHIR 3]] and link:http://hl7.org/fhir/R4/datatypes.html[FHIR 4]]
```
Comment (bitPogo) [inline DEVELOPMENT.adoc:151]: `link:http://hl7.org/fhir/STU3/datatypes.html[FHIR 3]`

Comment (bitPogo) [reply DEVELOPMENT.adoc:151]: `link:http://hl7.org/fhir/R4/datatypes.html[FHIR 4]`

File: DEVELOPMENT.adoc [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+Add:
+. New FHIR spec under `fhir-spec/hl7.org/fhir/{new FHIR version}/`
+. New `config` and `templates` under `fhir/parser/{new FHIR version}/`
+.. Adjust them to your needs
+. Base implementation
+.. FHIR model base `Fhir{new FHIR version}`
+.. FHIR primitives
+.. `Fhir{new FHIR version}Parser` and register it in `FhirParserFactory` alongside with it's JSON implementation
+. new FHIR version handling to `generateFhir.main.kts`
+. generated models, tests and validation resources by triggering generation
```
Comment (bitPogo) [inline DEVELOPMENT.adoc:178]: Also this should be bullet points, right? They are not shown correctly on GitHub.

File: README.adoc [hunk 5]
```diff
@@ -38,153 +40,199 @@ endif::[]
...
+
+[source, Kotlin]
 ----
+val parser = FhirParserFactory.createR4Parser()
 
-Make sure you are in the project root and install the correct Python version which is derived from `.python-version` file:
+val data = "JSON FHIR data"
 
-[source,bash]
+val result = parser.fromJson(DomainResource::class, "JSON FHIR data")
```
Comment (bitPogo) [inline README.adoc:153]: I guess this should be `data`, right otherwise `val data =  "JSON FHIR data"` is redundant.

File: README.adoc [hunk 6]
```diff
@@ -38,153 +40,199 @@ endif::[]
...
+FHIR Resources and Elements are represented as Kotlin interfaces which are named `Fhir` + ResourceType` e.g. `FhirDocumentReference`. While a `FhirDocumentReference` is implemented by `DocumentReference` which is a Kotlin data class. So they are equatable, copyable, destructable by default.
 
-=== First time
+[source, Kotlin]
+----
+val name = HumanName(family = "Doe", given = listOf("John"))
+val patient = Patient(name = listOf(name))
+
+val newName = HumanName(family = "Johnny", given = listOf("Silverhand"))
+val patientCopy = patient.copy(name = listOf(newName))
```
Comment (bitPogo) [inline README.adoc:180]: Could we use maybe a non-binary name like `Andi` or at least one female name here?

File: README.adoc [hunk 7]
```diff
@@ -38,153 +40,199 @@ endif::[]
...
+----
 
-As the project includes a Git submodule, this must be initialised on first use
+If you need to access the FHIR `resourceType` from a given `class` or need to get the `class` for a given FHIR `resourceType` use the `FhirHelper.FhirElementFactory`:
 
-[source,bash]
+[source, Kotlin]
 ----
-git submodule update --init --recursive
+val resourceType = FhirHelper.FhirElementFactory..getFhirResourceType(Patient::class)
```
Comment (bitPogo) [inline README.adoc:187]: `val resourceType = FhirHelper.FhirElementFactory.getFhirResourceType(Patient::class)`

**Review by bitPogo — APPROVED** [review]
Just some nitpicks.

**Review by bitPogo — CHANGES_REQUESTED** [review]
See above

File: DEVELOPMENT.adoc (hunk 2 above)
Comment (wmontwe) [reply DEVELOPMENT.adoc:147]: Should have been a checkbox, but will remove it

File: DEVELOPMENT.adoc (hunk 4 above)
Comment (wmontwe) [reply DEVELOPMENT.adoc:178]: I'll change it to list style

**Review by bitPogo — CHANGES_REQUESTED** [review]
One tiny nitpick is still open.

**Review by bitPogo — DISMISSED** [review]
LGTM

**Comment by wmontwe** [comment]
I had to downgrade Kotlin DateTime 0.2.1 -> 0.2.0 due to a bug introduced for iOS

**Review by bitPogo — APPROVED** [review]
LGTM

**Pull request merged**